lms/util/authorize_lms.py

```diff
@@ -67,17 +67,16 @@ def wrapper(request, *args, user=None, **kwargs):
             if oauth_condition(request) is False:
                 return view_function(request, *args, user=user, **kwargs)
 
+            ai_getter = request.find_service(name="ai_getter")
             consumer_key = request.params["oauth_consumer_key"]
-
-            application_instance = find_by_oauth_consumer_key(request.db, consumer_key)
-
-            client_id = application_instance.developer_key
-
-            if application_instance is None:
+            try:
+                client_id = ai_getter.developer_key(consumer_key)
+                lms_url = ai_getter.lms_url(consumer_key)
+            except ConsumerKeyError:
                 raise exc.HTTPInternalServerError()
```

Comment:
Does an invalid consumer key always indicate an internal error, as opposed to an error in the code making the request to the lms app?

Reply:
I think it probably _always_ indicates a problem with the request so should be a 4xx, but deliberately 500ing in this and a few other situations is the current behaviour of the app. Something to be fixed later. Possibly by adding validation on front of the views so that view code can assume that the consumer key param is valid.